### DISCUSSION TITLE
rebase with date=now

### DISCUSSION THREAD

**Discussion by annasun28** [arxanas/git-branchless#367] [Q&A] [answered]
Really love this project - I switched from working on a mercurial-based project to a git-based one recently, and this has made the change *much* nicer! 

One feature I am wondering about is whether it's possible to add the flag `--date=now` to all `git rebase` commands (which are called whenever I `git restack`, for e.g.). I'd like my commit history to be chronological/reflect the date that a rebase or commit --amend happened - I tried poking around in my gitconfig but it seems that an alias doesn't work for this. 

Is there an easy solution that I'm missing?

Thanks again for the great work!

**Comment by arxanas** [comment, marked as answer]
Hi @annasun28, I'm glad you find git-branchless useful! If you haven't already, it would be appreciated if you were to take [the user survey](https://github.com/arxanas/git-branchless/discussions/330) 🙂.

To clarify, you want the *author* dates to reflect the most-recently-modified time, not the *committer* dates? Can you tell me more about what your workflow around timestamps looks like?

The [`branchless.restack.preserveTimestamps`](https://github.com/arxanas/git-branchless/wiki/Configuration#branchlessrestackpreservetimestamps) option exists, but that does the opposite of what you want, it sounds like: it keeps both the author and committer dates in the past, rather than updating them to the present time.

If you're interested in adding an option to force the author timestamps to the present, it should be fairly straightforward, and I can walk you through it in this forum or on Discord. Here's an example of how we fix the timestamp to be in the past:

https://github.com/arxanas/git-branchless/blob/6a2ed25d8cf8d8b7adb9cf8681301b6220d90be6/git-branchless/src/commands/amend.rs#L96-L104

Oh, actually, from this code snippet, it looks like `git branchless amend` will already update the author time, which makes sense to me. For the rebase case (in `git move`/`git restack`/etc.), do you still want to update the author time? In my view, moving commits around doesn't usually change the author time, unless perhaps merge conflicts were resolved.

**Reply by annasun28** [reply]
Ah, you're right - I was incorrectly looking at author timestamps; indeed the committer timestamps are updated to date=now after  (based on the default preserveTimestamps=False), which is what I wanted. Thank you for the clarification on the distinction!
